### ISSUE TITLE
Gradient clipping support

### ISSUE THREAD

**Issue by petrbel** [Cognexa/cxflow-tensorflow#77]
`BaseModel._create_train_ops` should be configurable to perform _gradient clipping_. This will allow us to implement and train more advanced models such as seq2seq etc.
